packages/kafka/lib/AbstractKafkaConsumer.ts

```diff
@@ -211,12 +211,20 @@ export abstract class AbstractKafkaConsumer<
       )
       this.messageBatchStream.on('error', (error) => this.handlerError(error))
     } else {
-      this.consumerStream.on('data', (message) =>
-        this.consume(
-          message.topic,
-          message as DeserializedMessage<SupportedMessageValues<TopicsConfig>>,
-        ),
-      )
+      // biome-ignore lint/style/noNonNullAssertion: consumerStream is always created
+      const stream = this.consumerStream!
+      stream.on('data', async (message) => {
+        stream.pause()
```

Comment:
this prevents any kind of parallelism, processing only one at a time.

what you think about something like this?

```ts
const MAX_IN_FLIGHT = 10 // should be a configureable value
let inFlight = 0

this.consumerStream.on('data', async (message) => {
  inFlight++

  if (inFlight >= MAX_IN_FLIGHT) {
    this.consumerStream.pause()
  }

  try {
    await this.consume(
      message.topic,
      message as DeserializedMessage<SupportedMessageValues<TopicsConfig>>,
    )
  } finally {
    inFlight--

    // If we were paused and now we have room, resume
    if (inFlight < MAX_IN_FLIGHT) {
      this.consumerStream.resume()
    }
  }
})
```